syncer/checkpoint.go

```diff
@@ -226,28 +229,30 @@ func NewRemoteCheckPoint(tctx *tcontext.Context, cfg *config.SubTaskConfig, id s
 }
 
 // Init implements CheckPoint.Init
-func (cp *RemoteCheckPoint) Init(conn *Conn) error {
+func (cp *RemoteCheckPoint) Init(conn *DBConn) error {
 	if conn != nil {
-		cp.db = conn
+		cp.dbConn = conn
 	} else {
-		db, err := createConn(cp.cfg, cp.cfg.To, maxCheckPointTimeout)
+		cp.cfg.To.RawDBCfg = config.DefaultRawDBConfig(maxCheckPointTimeout)
```

Comment:
this line will overwrite `RawDBCfg` used in other places because `cp.cfg` is a _pointer_.

Reply:
I fixed it [here](https://github.com/pingcap/dm/pull/266/files#diff-ef22d46ca0482ca5b7cf503016990902R233), and I also fixed [`onlineDDL`](https://github.com/pingcap/dm/pull/266/files#diff-18d840e544a27bd6ebe280a590e1dbf9R111), [`shardGroup`](https://github.com/pingcap/dm/pull/266/files#diff-013e2adce506b91024f15af06c7f7027R462)